modules/contacts.js

```diff
@@ -0,0 +1,231 @@
+/**
+ * @access public
+ * @description this contact backup and restore functionality has the ability to look on your device and the contacts that are stored on the SIM or ICC card(s) as well as internal contacts, and back them up to a JSON file on an external SD card.
+ */
+var Contacts = function() {
+  this.contacts = [];
+};
+
+/**
+ * @access public
+ * @description TODO
+ */
+Contacts.prototype.backup = function() {
+  this.getContactsFromSIM();
+};
+
+/**
+ * @access public
+ * @description The Restore function will look for a contacts JSON file existing on an extenal SD card under the directory /backup/contacts/contacts.json. It will parse all the data in that file and transfer them back onto the internal memory in mozContacts
+ */
+Contacts.prototype.restore = function() {
+  console.log('Restoring contacts.');
+
+  var that = this;
+  var reader = new FileReader();
+
+  reader.onloadend = function() {
+    var contents = this.result;
+    var data = JSON.parse(contents);
+    for (var i = 0; i < data.length; ++i) {
+      var myContact = new mozContact(data[i]);
+      myContact.givenName = [data[i].name];
```

Comment:
If givenName and familyName aren't set, use name. Otherwise, populate both fields